tools/toolchains/arm.py

```diff
@@ -66,6 +66,9 @@ def __init__(self, target, notify=None, macros=None,
         self.asm = [main_cc] + self.flags['common'] + self.flags['asm']
         self.cc = [main_cc] + self.flags['common'] + self.flags['c']
         self.cppc = [main_cc] + self.flags['common'] + self.flags['c'] + self.flags['cxx']
+        if target.name in ["REALTEK_RTL8195AM"]:
```

Comment:
Indeed, enum was typedef-ed to a new type and passed it across modules. Part of the readonly code was built with enum size of 4, the rest just has to comply. 

I figured out a way to fix enum size issue. Instead of typedef enum to a new type and force it to be 4-byte. The better way is to leave enum as it is, and add additional typedef to int or uint32_t (depends on the enum characteristics). That way, enum itself is not changed, and the size of fixed to 4 across compilers. The modification is minimal, only need to modify a couple of header files.

Example:

`typedef enum {
  .... 
  ....
} UART_State, *PUART_State`

`enum {
.... 
....
};
typedef uint32_t UART_State;
typedef uint32_t *PUART_State;
`